dpctl/tensor/libtensor/include/kernels/elementwise_functions/imag.hpp

```diff
@@ -70,11 +71,13 @@ template <typename argT, typename resT> struct ImagFunctor
     resT operator()(const argT &in) const
     {
         if constexpr (is_complex<argT>::value) {
-            return std::imag(in);
+            using realT = typename argT::value_type;
+            using sycl_complexT = typename exprm_ns::complex<realT>;
+            return exprm_ns::imag(sycl_complexT(in));
```

Comment:
It might be unclear from the PR description what was the motivation to switch to experimental namespace and why other place where `std::imag` used are not impacted.. Would it make sense to add clarity on that?

Reply:
The initial motivation was old technical debt to add the constant value for `imag`, since we may as well use it for real-valued types.

In terms of why add in the experimental namespace change, `complex` header is not supported on AMD and this has been an issue with recent AMD builds. I've pushed another branch which removes (almost) every use of `complex` in kernel code aside from conversion to SYCL complex type. I want to see if this resolves issues with AMD builds on OS compiler.

Reply:
In general, I think it would be best to slowly transition to full use of the sycl complex extension. We also have benchmarking now and can see if this PR brings regressions.

Reply:
I see now, thank you. I'm totally with that. Just it was a bit unclear from the PR description.